Google.Api.Gax.Grpc/RetrySettings.cs

```diff
@@ -161,11 +163,21 @@ public static Predicate<RpcException> FilterForStatusCodes(IEnumerable<StatusCod
         }
 
         /// <summary>
-        /// Builds a new RetrySettings which is identical to this one, but with the given expiration.
+        /// Works out the next backoff from the current one, based on the multiplier and maximum.
         /// </summary>
-        /// <param name="expiration">New expiration</param>
-        public RetrySettings WithTotalExpiration(Expiration expiration) =>
-            new RetrySettings(RetryBackoff, TimeoutBackoff, expiration, RetryFilter, DelayJitter);
+        /// <param name="currentBackoff">The current backoff to use as a basis for the next one.</param>
+        /// <returns>The next backoff to use, which is always at least <see cref="InitialBackoff"/> and at most <see cref="MaxBackoff"/>.</returns>
+        public TimeSpan NextBackoff(TimeSpan currentBackoff)
```

Comment:
A little late to the party, but I would expect this method to use whatever jitter is set.
I know the jitter is being applied when retrying, but now that the backoff configuration and the jitter are defined on the same class, it's just unexpected that this doesn't use the jitter.
I would at least add a comment here.
And I would probably actually add a new property `InitialJitteredBackoff` and a new method `NextJitteredBackoff`. The retry code would be slightly simpler with these new members.

Reply:
I don't think it would make sense to have extra properties (what would `InitialJitteredBackoff` mean?) - but a few options:

- Just add documentation
- Add an `out` parameter for the "jittered backoff"
- Write a method that encapsulates all of this by returning an `IEnumerable<TimeSpan>` that is "actual backoffs to apply" and which keeps track of the multiplied version

Thoughts?

Reply:
I really like option 3, as long as the existing exposed methods and properties are not removed so that client code can use the retry config on other delays that were not generated in the sequence (for instance if the server recommends a delay (I know this doesn't happen now)).
Options 2 is good enough, only problem would be that the initial backoff still needs to be jittered "manually" (that's what the InitialJitteredBackoff property does).

Reply:
How about option 3, but removing the existing method?

For option 2, I'd envisaged it being "the out parameter is how much you should delay right now, and the return value is what should be passed in next time". So there wouldn't be any need for manual jittering.

Reply:
I like option 3 (sequence) with existing method removed. We can always add it back if a use-case for it arises.